app/controllers/gift_requests_controller.rb

```diff
@@ -7,20 +7,24 @@ class GiftRequestsController < ApplicationController
 
   def new
     @back_route = wishlist_path(params[:id])
-
     @gift_request = GiftRequest.new
-    render "gift_request", locals: {list_id: params[:id]}
+    @list_name = List.find(params[:id]).name
+
+    render "gift_request"
   end
 
   def edit
     @back_route = wishlist_path(params[:id])
-
-    render "gift_request", locals: {list_id: params[:id]}
+    @list_name = List.find(params[:id]).name
+    
+    render "gift_request"
   end
 
   def create
     # params[:id] is the list id this gift request corresponds to
     @gift_request = GiftRequest.new(gift_request_params.merge(list_id: params[:id]))
+    @list_name = List.find(params[:id]).name
```

Comment:
In the case of `create`, I can't get the list name through the gift request. Either the gift request hasn't been created yet or the creation of the gift request failed (in both cases there is no associated list). Let me know if there's a better way to handle this